opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1000,21 +1001,32 @@ namespace Opm {
                 fip_.fip[i].resize(nc,0.0);
             }
 
-            for (int c = 0; c < nc; ++c) {
-                const auto& intQuants = *ebosSimulator_.model().cachedIntensiveQuantities(c, /*timeIdx=*/0);
+            ElementContext elemCtx(ebosSimulator_);
+            auto elemIt = elemCtx.gridView().template begin</*codim=*/0>();
+            const auto& elemEndIt = elemCtx.gridView().template end</*codim=*/0>();
+            for (; elemIt != elemEndIt; ++elemIt) {
```

Comment:
ok. with `Dune::CpGrid` it does not work: it says

```
error: ‘elements’ was not declared in this scope
```

the same story applies if `Dune::elements()` is used. that seems to be an issue with `Dune::CpGrid`? (also, would it work with Dune 2.3? If no, this could be a good reason to raise the Dune requirement to 2.4...)